state/errors.go

```diff
@@ -72,6 +74,34 @@ func IsOutdatedExtensionError(err error) bool {
 	return errors.As(err, &OutdatedExtensionError{})
 }
 
+// UnverifiableExtensionError represents a state extension (block) which cannot be
+// verified at the moment. For example, it does not connect to the finalized state,
+// or an entity referenced within the payload is unknown.
+// Unlike InvalidExtensionError, this error is only used when the failure CANNOT be
+// attributed to a malicious input, therefore this error can be treated as a benign failure.
+type UnverifiableExtensionError struct {
+	err error
+}
```

Comment:
A comment that can be ignored: I think embedding the error avoids the need of adding the `Error(). string` method. 
```
type UnverifiableExtensionError struct {
	error
}
```

Reply:
Good point! Refactored all the `state/*` errors in https://github.com/onflow/flow-go/pull/3577/commits/d0bd5c375fceb6a63bfa7af09a9f5021c49b0a20